uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/calendar-ctrl.js

```diff
@@ -0,0 +1,283 @@
+define([
+  'leave-absences/shared/modules/controllers',
+  'common/lodash',
+  'common/moment',
+  'leave-absences/shared/models/absence-period-model',
+  'leave-absences/shared/models/absence-type-model',
+  'leave-absences/shared/models/public-holiday-model',
+], function (controllers, _, moment) {
+  'use strict';
+
+  controllers.controller('CalendarCtrl', ['$timeout', 'shared-settings', 'AbsencePeriod', 'AbsenceType',
+    'PublicHoliday', 'OptionGroup', controller]);
+
+  function controller($timeout, sharedSettings, AbsencePeriod, AbsenceType, PublicHoliday, OptionGroup) {
+    this._dayTypes = [];
+    this._publicHolidays = [];
+    this._leaveRequestStatuses = [];
+
+    this.absencePeriods = [];
+    this.absenceTypes = [];
+    this.months = [];
+    this.selectedMonths = [];
+    this.selectedPeriod = null;
+    this.loading = {
+      calendar: false,
+      page: false
+    };
+    this.monthLabels = ['January', 'February', 'March', 'April', 'May', 'June',
+      'July', 'August', 'September', 'October', 'November', 'December'];
+
+    /**
+     * Fetch months from newly selected period and refresh data
+     */
+    this.changeSelectedPeriod = function () {
+      this._fetchMonthsFromPeriod();
+      this.refresh();
+    };
+
+    /**
+     * Labels the given period according to whether it's current or not
+     *
+     * @param  {object} absenceType
+     * @return {object} style
+     */
+    this.getAbsenceTypeStyle = function (absenceType) {
+      return {
+        backgroundColor: absenceType.color,
+        borderColor: absenceType.color
+      };
+    };
+
+    /**
+     * Returns day name of the sent date(Monday, Tuesday etc.)
+     *
+     * @param  {string} date
+     * @return {string}
+     */
+    this.getDayName = function (date) {
+      var day = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
+      return day[this._getDateObjectWithFormat(date).day()];
+    };
+
+    /**
+     * Decides whether sent date is a public holiday
+     *
+     * @param  {string} date
+     * @return {boolean}
+     */
+    this.isPublicHoliday = function (date) {
+      return !!this._publicHolidays[this._getDateObjectWithFormat(date).valueOf()];
+    };
+
+    /**
+     * Labels the given period according to whether it's current or not
+     *
+     * @param  {AbsencePeriodInstance} period
+     * @return {string}
+     */
+    this.labelPeriod = function (period) {
+      return period.current ? 'Current Period (' + period.title + ')' : period.title;
+    };
+
+    /**
+     * Converts given date to moment object with server format
+     *
+     * @param {Date/String} date from server
+     * @return {Date} Moment date
+     */
+    this._getDateObjectWithFormat = function (date) {
+      return moment(date, sharedSettings.serverDateFormat).clone();
```

Comment:
It was done to avoid a specific error which I cant remember. Also after removing it I cant seems to find any problems. So removed.